Assets/Mirror/Tests/MessageBaseTests.cs

```diff
@@ -0,0 +1,51 @@
+using NUnit.Framework;
+
+namespace Mirror
+{
+    struct TestMessage : IMessageBase
+    {
+        public int IntValue;
+        public string StringValue;
+        public double DoubleValue;
+
+        public TestMessage(int i, string s, double d)
+        {
+            IntValue = i;
+            StringValue = s;
+            DoubleValue = d;
+        }
+
+        public void Deserialize(NetworkReader reader)
+        {
+            IntValue = reader.ReadInt32();
+            StringValue = reader.ReadString();
+            DoubleValue = reader.ReadDouble();
+        }
+
+        public void Serialize(NetworkWriter writer)
+        {
+            writer.Write(IntValue);
+            writer.Write(StringValue);
+            writer.Write(DoubleValue);
+        }
+    }
+
+    [TestFixture]
+    public class MessageBaseTests
+    {
+        [Test]
+        public void Roundtrip()
+        {
+            NetworkWriter w = new NetworkWriter();
+            w.Write(new TestMessage(1, "2", 3.3));
+
+            var arr = w.ToArray();
+
+            var r = new NetworkReader(arr);
+            var t = new TestMessage();
```

Comment:
There's still some vars here, sorry I positioned the comment wrong last time. Also these one letter name should be more descriptive like `writer`, `reader`, `testMessage` etc.

Reply:
Doh, I should probably have noticed those when making the edits!